src/sdl2_compat.c

```diff
@@ -7698,32 +7698,36 @@ SDL_GetDesktopDisplayMode(int displayIndex, SDL2_DisplayMode *mode)
 
 #define PROP_WINDOW_FULLSCREEN_MODE "sdl2-compat.window.fullscreen-mode"
 
-static int
-ApplyFullscreenMode(SDL_Window *window)
+static int ApplyFullscreenMode(SDL_Window *window)
 {
-    SDL2_DisplayMode *property = (SDL2_DisplayMode *) SDL3_GetPointerProperty(SDL3_GetWindowProperties(window), PROP_WINDOW_FULLSCREEN_MODE, NULL);
+    /* Always try to enter fullscreen on the current display */
+    const SDL_DisplayID displayID = SDL3_GetDisplayForWindow(window);
+    SDL2_DisplayMode *property = (SDL2_DisplayMode *)SDL3_GetPointerProperty(SDL3_GetWindowProperties(window), PROP_WINDOW_FULLSCREEN_MODE, NULL);
     SDL_DisplayMode mode;
+
     SDL3_zero(mode);
     if (property) {
-        /* Always try to enter fullscreen on the current display */
-        const SDL_DisplayID displayID = SDL3_GetDisplayForWindow(window);
-
-        /* The SDL2 refresh rate is rounded off, and SDL3 checks that the mode parameters match exactly, so try to find the closest matching SDL3 mode. */
-        SDL3_GetClosestFullscreenDisplayMode(displayID, property->w, property->h, (float)property->refresh_rate, false, &mode);
+        mode.w = property->w;
+        mode.h = property->h;
+        mode.refresh_rate = (float)property->refresh_rate;
+    } else {
+        const SDL_DisplayMode *desktop_mode = SDL3_GetDesktopDisplayMode(displayID);
```

Comment:
I don't think this part is correct. SDL2 uses similar logic to our `SDL_GetWindowDisplayMode()` to determine the fullscreen resolution, but it will fail outright if a suitable resolution can't be found instead of picking desktop mode.

Reply:
Ah, never mind. I saw that you are overriding the dimensions with the window size. Though, I think calling `SDL3_GetDesktopDisplayMode()` isn't necessary  because `SDL3_GetClosestFullscreenDisplayMode()` will use the desktop refresh rate already if you pass 0.0f.

Reply:
Good call, I'll remove that.